src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -469,14 +473,17 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
             if (project.Language != LanguageNames.CSharp &&
                 project.Language != LanguageNames.VisualBasic)
             {
-                return;
+                return null;
             }
 
             // Project was changed in some way.  Let's go find the set of installed packages for it.
-            var dteProject = _workspace.TryGetDTEProject(projectId);
+            return _workspace.TryGetDTEProject(projectId);
+        }
+
+        private void ProcessProjectChange(EnvDTE.Project dteProject, ProjectId projectId)
+        {
             if (dteProject == null)
```

Comment:
1. this should have a threading assert.
2. below, you call into _packageService.GetInstalledPackages.  how do you know that is safe to call on this thread?